linera-base/src/data_types/balance_scalar.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A custom `ScalarType` implementation for `Balance`.
+//!
+//! GraphQL doesn't support `u128`, so we represent a balance as a list with two elements, the most
+//! and least significant 64 bits.
+//!
+//! This is mostly adapted from the implementation of the `async_graphql::scalar!` macro.
+
+use std::{borrow::Cow, iter, sync::Arc};
+
+use async_graphql::{
+    parser::types::Field,
+    registry::{MetaType, MetaTypeId, Registry},
+    ContextSelectionSet, InputType, InputValueError, InputValueResult, Name, OutputType,
+    Positioned, ScalarType, ServerResult, Value,
+};
+
+use super::Balance;
```

Comment:
Nit: maybe remove the newlines separating the imports? Rustfmt will then reorder it to:

```suggestion
use super::Balance;
use async_graphql::{
    parser::types::Field,
    registry::{MetaType, MetaTypeId, Registry},
    ContextSelectionSet, InputType, InputValueError, InputValueResult, Name, OutputType,
    Positioned, ScalarType, ServerResult, Value,
};
use std::{borrow::Cow, iter, sync::Arc};
```

Reply:
Yes, please!